### PR TITLE
Remove hibernate entity hydration from logObservation critical path

### PR DESCRIPTION
**Pull request by coryverta**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java [hunk 1]
```diff
@@ -846,20 +861,30 @@ public void logObservations(String experimentRunId, List<Observation> observatio
...
-          RdbmsUtils.convertObservationsFromObservationEntityList(
-              session,
-              experimentRunEntityObj,
-              ModelDBConstants.OBSERVATIONS,
-              observations,
-              ExperimentRunEntity.class.getSimpleName(),
-              experimentRunEntityObj.getId());
-      experimentRunEntityObj.setObservationMapping(newObservationList);
-      long currentTimestamp = Calendar.getInstance().getTimeInMillis();
-      experimentRunEntityObj.setDate_updated(currentTimestamp);
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java:861]: Update the `date_updated` field in experiment run is missing in the new code

File: backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java [hunk 2]
```diff
@@ -2709,19 +2763,26 @@ public Response commitArtifactPart(CommitArtifactPart request) throws ModelDBExc
...
-    query.setParameter(ModelDBConstants.EXPERIMENT_RUN_ID_STR, experimentRunId);
-    query.setParameter("field_type", fieldType);
-    query.executeUpdate();
+    session
+        .createNativeQuery(
+            "DELETE FROM keyvalue WHERE experiment_run_id = :experiment_run_id"
+                + " AND field_type = :field_type")
+        .setParameter("experiment_run_id", experimentRunId)
+        .setParameter("field_type", fieldType)
+        .executeUpdate();
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java:2773]: this is the same as old code, i do not understand what is the difference on both and purpose??

File: backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java [hunk 3]
```diff
@@ -2709,19 +2763,26 @@ public Response commitArtifactPart(CommitArtifactPart request) throws ModelDBExc
...
-    query.executeUpdate();
+    session
+        .createNativeQuery(
+            "DELETE FROM keyvalue WHERE experiment_run_id = :experiment_run_id"
+                + " AND field_type = :field_type"
+                + " AND kv_key IN (:keys)")
+        .setParameter("experiment_run_id", experimentRunId)
+        .setParameter("field_type", fieldType)
+        .setParameter("keys", keys)
+        .executeUpdate();
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java:2786]: this is the same as old code, i do not understand what is the difference on both and purpose??

File: backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java [hunk 4]
```diff
@@ -2775,34 +2832,31 @@ public void deleteExperimentRunObservationsEntities(
...
+                    " AND (oe.keyValueMapping.key IN (:keys) OR oe.artifactMapping.key IN (:keys))")
+                .toString();
+        Query query = session.createQuery(GET_SELECTED_OBSERVATIONS_HQL);
+        query.setParameter(ModelDBConstants.EXPERIMENT_RUN_ID_STR, experimentRunId);
+        query.setParameter("field_type", ModelDBConstants.OBSERVATIONS);
+        query.setParameterList("keys", experimentRunObservationsKeys);
+        List<ObservationEntity> observationEntities = query.list();
+        for (ObservationEntity observationEntity : observationEntities) {
+          session.delete(observationEntity);
+        }
```
Comment (ad-47) [inline backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java:2857]: this all changes same as i have did it in my PR right?

File: backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java:861]: Need to be add `updateExperimentRunTimestamp(experimentRunId, session);`

File: backend/src/main/java/ai/verta/modeldb/monitoring/RequestLatencyResource.java [hunk 5]
```diff
@@ -9,6 +9,9 @@
           .labelNames("grpc_method")
           .name("verta_backend_requests_latency_seconds")
           .help("Request latency in seconds.")
+          .buckets(
+              0.2, 0.4, 0.6, 0.8, 1.0, 1.2, 1.4, 1.6, 1.8, 2.0, 2.2, 2.4, 2.6, 2.8, 3.0, 3.2, 3.4,
+              3.6, 3.8, 4.0)
```
Comment (anandjakhaniya) [inline backend/src/main/java/ai/verta/modeldb/monitoring/RequestLatencyResource.java:14]: What about this???

**Review by anandjakhaniya — CHANGES_REQUESTED** [review]
Small comment on `date_updated` missing at one place.

**Comment by anandjakhaniya** [comment]
This PR build on top of my PR https://github.com/VertaAI/modeldb/pull/2046 right??

File: backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java (hunk 2 above)
Comment (coryverta) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java:2773]: Native queries bypass the hibernate session cache and statement cache.  This removes the extra processing required by hibernate to translate HQL into SQL, and prevents hibernate from caching the query or the results. 

Under high load, the extra processing time required by hibernate to translate and cache the query become a bottleneck.

File: backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java (hunk 3 above)
Comment (coryverta) [reply backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java:2786]: See above

File: backend/src/main/java/ai/verta/modeldb/monitoring/RequestLatencyResource.java (hunk 5 above)
Comment (coryverta) [reply backend/src/main/java/ai/verta/modeldb/monitoring/RequestLatencyResource.java:14]: The default histogram for prometheus only uses 10 buckets with a wide spread.  This bucket distribution makes it much easier to see fine-grained changes in the performance data.

**Comment by coryverta** [comment]
@ad-47 Yes, I created this branch from your PR.

**Pull request merged**